applications/HDF5Application/custom_utilities/data_type_utilities.h

```diff
@@ -88,7 +90,89 @@ void GetShapeFromAttributes(
     }
 }
 
+template<class TContainerType>
+[[nodiscard]] std::vector<std::string> GetListOfAvailableVariables(
+    TContainerType& rContainer,
+    const DataCommunicator& rDataCommunicator)
+{
+    KRATOS_TRY
+
+    using map_type = std::unordered_map<VariableData::KeyType, std::string>;
+
+    // Map to Map reduction class
+    class MapReduction
+    {
+    public:
+        using return_type = map_type;
+
+        return_type mValue;
+
+        /// access to reduced value
+        return_type GetValue() const
+        {
+            return mValue;
+        }
```

Comment:
`GetValue` only gets called once, and the reduction instance gets discarded afterwards, so we can `move` the member out instead of copying it.

```suggestion
        [[nodiscard]] return_type&& GetValue()
        {
            return std::move(mValue);
        }
```